src/bsp/platform/iotc_bsp_io_net_test.cc

```diff
@@ -76,28 +77,32 @@ bool ServerTest::WaitUntilSocketReadyForRead() {
 }
 
 TEST_P(ServerTest, EndToEndCommunicationWorks) {
-  const NetworkType* test_case = GetParam();
-  socket_type_ = test_case->socket_type;
-  listening_addr_ = test_case->listening_addr;
-  protocol_type_ = test_case->protocol_type;
+  const NetworkType test_case = GetParam();
+  const char* kTestMsg = "hello\n";
+  const uint16_t kTestPort = 2000;
+  std::string listening_addr_ = test_case.listening_addr;
+  iotc_bsp_socket_type_t socket_type_ = test_case.socket_type;
+  iotc_bsp_protocol_type_t protocol_type_ = test_case.protocol_type;
+  int out_written_count_;
 
-  std::unique_ptr<EchoTestServer> test_server(
-      new EchoTestServer(socket_type_, kTestPort, protocol_type_));
+  std::unique_ptr<EchoTestServer> test_server =
```

Comment:
You should do:

`auto test_server = std::make_unique<EchoTestServer>(...);`

because the type of `test_server` is implicit. The code stays readable, and is simpler, because you don't have to repeat the type twice.